influxql/ast.go

```diff
@@ -952,6 +952,16 @@ func (s *SelectStatement) RewriteWildcards(ic IteratorCreator) (*SelectStatement
 
 	// If there are no dimension wildcards then merge dimensions to fields.
 	if !hasDimensionWildcard {
+		// Remove the dimensions present in the group by so they don't get added as fields.
+		for _, d := range s.Dimensions {
+			switch expr := d.Expr.(type) {
```

Comment:
nit: is there likely to be any other type we need to switch on in the future? And if not would  `if expr, ok := d.Expr.(*VarRef); ok {` maybe be better?